api/scpca_portal/models/library.py

```diff
@@ -104,3 +109,45 @@ def get_formats_from_file_paths(cls, file_paths: List[Path]) -> List[str]:
         if any(path for path in file_paths if common.ANNDATA_EXT == path.suffix):
             formats.append(Library.FileFormats.ANN_DATA)
         return formats
+
+    @classmethod
+    def get_project_libraries_from_download_config(
+        cls, project, download_configuration: Dict
+    ) -> List[Self]:
+        if download_configuration not in common.GENERATED_PROJECT_DOWNLOAD_CONFIGURATIONS:
+            raise ValueError("Invalid download configuration passed. Unable to retrieve libraries.")
+
+        if download_configuration["includes_merged"]:
+            # If the download config requests merged and there is no merged file in the project,
+            # return an empty list
+            if (not project.includes_merged_sce) and (not project.includes_merged_anndata):
+                return []
```

Comment:
This is too broad a check, we need to check if the specific merged data is available for the format in the `download_configuration`. So we only check `includes_merged_sce` when `format` is `SINGLE_CELL_EXPERIMENT` and alternatively check `includes_merged_anndata` when `format` is `ANN_DATA`.

This probably wouldn't break anything today but would be a bug should be have a project that is merged in one format but not another.